### PR TITLE
etcdserver: set applied index correctly

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/5950

/cc @gyuho

### CODE REVIEW

File: etcdserver/server.go [hunk 1]
```diff
@@ -1044,6 +1044,7 @@ func (s *EtcdServer) applyEntryNormal(e *raftpb.Entry) {
 		s.consistIndex.setConsistentIndex(e.Index)
 		shouldApplyV3 = true
 	}
+	s.setAppliedIndex(e.Index)
```
Comment (heyitsanthony) [inline etcdserver/server.go]: I think this should be a `defer`? Otherwise it's updated before the apply call (if any).

Comment (xiang90) [inline etcdserver/server.go]: ok. sure.

**Comment by xiang90** [comment]
@heyitsanthony Fixed.

**Comment by gyuho** [comment]
lgtm. thanks!

**Comment by gyuho** [comment]
@xiang90 Could you explain how this fixes that ErrTooManyRequests issue for me? Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by xiang90** [comment]
@gyuho This does not need to be back-ported. We recently added a new mechanism to ensure the max gap between committed index and applied index, so that we wont submit proposal faster than we can apply. See https://github.com/coreos/etcd/pull/5927. We need to ensure the applied index is calculated correctly.

**Comment by gyuho** [comment]
@xiang90 Got it. Thanks!

**Pull request merged**